api/ResponseTypes.ts

```diff
@@ -0,0 +1,122 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+export interface Search<T = any> {
```

Comment:
I would recommend `T = unknown` instead of `T = any`: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type

Reply:
I'm using `any` for consistency with the `RequestParams`, change that now would lead to a breaking change, unfortunately.